core/test/junit/saros/concurrent/SplitOperationTest.java

```diff
@@ -58,6 +57,93 @@ public void testInsertInsert() {
     assertEquals(Collections.singletonList(expected2), split2.toTextEdit(path, source));
   }
 
+  @Test
+  public void testMultiLineInsertInsertCase1() {
+    // Ins((0,4),"0ab\n") + Ins((1,0),"cd") -> Ins((0,4),"0ab\ncd")
+    Operation split1 = S(I(4, "0ab" + EOL), I(1, 0, "cd"));
+    TextEditActivity expected1 = T(source, 4, "0ab" + EOL + "cd", "", path);
+    assertEquals(Collections.singletonList(expected1), split1.toTextEdit(path, source));
+
+    // Ins((1,4),"0ab\ncd") + Ins((2,2),"efg") -> Ins((1,4),"0ab\ncdefg")
+    Operation split2 = S(I(1, 4, "0ab" + EOL + "cd"), I(2, 2, "efg"));
+    TextEditActivity expected2 = T(source, 1, 4, "0ab" + EOL + "cdefg", "", path);
+    assertEquals(Collections.singletonList(expected2), split2.toTextEdit(path, source));
+
+    // Ins((3,4),"0ab") + Ins((3,7),"cd\n") -> Ins((3,4),"0abcd\n")
+    Operation split3 = S(I(3, 4, "0ab"), I(3, 7, "cd" + EOL));
+    TextEditActivity expected3 = T(source, 3, 4, "0abcd" + EOL, "", path);
+    assertEquals(Collections.singletonList(expected3), split3.toTextEdit(path, source));
+
+    // Ins((0,4),"0ab") + Ins((0,7),"cd\nef") -> Ins((0,4),"0abcd\nef")
+    Operation split4 = S(I(4, "0ab"), I(7, "cd" + EOL + "ef"));
+    TextEditActivity expected4 = T(source, 4, "0abcd" + EOL + "ef", "", path);
+    assertEquals(Collections.singletonList(expected4), split4.toTextEdit(path, source));
+
+    // Ins((0,4),"0ab\ncd") + Ins((1,2),"ef\ngh") -> Ins((0,4),"0ab\ncdef\ngh")
+    Operation split5 = S(I(4, "0ab" + EOL + "cd"), I(1, 2, "ef" + EOL + "gh"));
+    TextEditActivity expected5 = T(source, 4, "0ab" + EOL + "cdef" + EOL + "gh", "", path);
+    assertEquals(Collections.singletonList(expected5), split5.toTextEdit(path, source));
+
+    // Ins((9,4),"0ab\ncd\nef\nghx") + Ins((12,3),"ij\nkl\nmn\nop") ->
+    // Ins((9,4),"0ab\ncd\nef\nghxij\nkl\mn\nop")
+    Operation split6 =
+        S(
+            I(9, 4, "0ab" + EOL + "cd" + EOL + "ef" + EOL + "ghx"),
+            I(12, 3, "ij" + EOL + "kl" + EOL + "mn" + EOL + "op"));
+    TextEditActivity expected6 =
+        T(
+            source,
+            9,
+            4,
+            "0ab" + EOL + "cd" + EOL + "ef" + EOL + "ghxij" + EOL + "kl" + EOL + "mn" + EOL + "op",
+            "",
+            path);
+    assertEquals(Collections.singletonList(expected6), split6.toTextEdit(path, source));
+  }
+
+  @Test
+  public void testMultiLineInsertInsertCase2() {
```

Comment:
It is not really related to this test case, but what happens if a Saros user writes a "\n" in his/her shared document? Is this detected as linebreak or is it escaped?